### DISCUSSION TITLE
On Demand revalidation with fallback: blocking. Only revalidate when page is already in cache

### DISCUSSION THREAD

**Discussion by wesleyjanse** [vercel/next.js#41594] [Ideas]
### Describe the feature you'd like to request

We work on a large e-Commerce website with 20k products, we staticly render these pages with long revalidation timers. And we let our backend revalidate pages when data changes.

Our customer does nightly syncs of these products which will update most of them, triggering revalidations for all pages. 

We would like the option that the revalidate, firstly checks the cache before revalidating and generating the page since we don't require these to be pre-rendered OR only purge the cache of the page and let fallback: blocking handle the rest. 

this will mitigate uneccesary load that is not needed.

### Describe the solution you'd like

- Only revalidate pages/urls that are allready in cache
- Do not generate pages via revalidate if the page doesn't exist in cache in the first place

Another cool option for us would be
- Just purge the cache, don't regenerate, since we pre-render nothing. We are used to the first SSR load of the page (which is fast enough) 
### Describe alternatives you've considered

- We are opting to mount a shared drive on our EB on AWS
- So we only need to overload 1 node, but this is not cost efficient since you never want to overload a specific node

**Comment by JonathanLauwers** [comment]
+1

**Comment by ijjk** [comment]
Hi, there is an `unstable_onlyGenerated: true` flag that can be passed e.g. `res.revalidate('/blog/post-1', { unstable_onlyGenerated })` does the behavior of that experimental config address your use case?

**Reply by wesleyjanse** [reply]
Hi, I think it does help a bit! But does onlyGenerated account for the pages that are pre-generated or also to the pages that are SSR generated via fallback blocking?
